### PR TITLE
Rename macros in config.h.

### PR DESCRIPTION
**Pull request by zhulipeng**
This patch try to do following things:
1. Rename `redis_*` and `REDIS_*` macros defined in config.h to `valkey_*`, `VALKEY_*` and update associated used files. (`redis_fstat`, `redis_fsync`, `REDIS_THREAD_STACK_SIZE`, etc.)
2. Remove the leading double underscore for guard macro in config.h.

### CODE REVIEW

File: src/hyperloglog.c [hunk 1]
```diff
@@ -393,7 +393,7 @@ static char *invalid_hll_err = "-INVALIDOBJ Corrupted HLL object detected";
 /* Our hash function is MurmurHash2, 64 bit version.
  * It was modified for Redis in order to provide the same result in
```
Comment (hwware) [inline src/hyperloglog.c]:
```suggestion
 * It was modified for Valkey in order to provide the same result in
```

**Review by hwware — COMMENTED** [review]
Just a minor comment, Thanks

File: src/hyperloglog.c (hunk 1 above)
Comment (zhulipeng) [reply src/hyperloglog.c]: Sure, I will update the comments in the changed files at once.

**Comment by zhulipeng** [comment]
@hwware @zuiderkwast Part of https://github.com/valkey-io/valkey/issues/25

**Review by zuiderkwast — APPROVED** [review]
We discussed earlier that it's better to make functions brand neutral, like using `server_` prefix instead of `valkey_`, but `server_fsync` would be a bit non-unique maybe. I don't mind that we keep `valkey` here for now.

~~Can you update the top comment to list all the names you search-replaced using regex? (If it's not too much...)~~ [Edit] I edited the top comment a little and used it as the commit message when merging.

Thanks!

**Pull request merged**